examples/cpp/pyperf/PyPerfBPFProgram.cc

```diff
@@ -489,6 +492,10 @@ get_first_arg_name(
   if (result == 0 && ob_size > 0) {
     result |= bpf_probe_read_user(&args_ptr, sizeof(void*), args_ptr + offsets->PyTupleObject.ob_item);
     result |= bpf_probe_read_user_str(argname, maxlen, args_ptr + offsets->String.data);
+    if (result < 0) {
```

Comment:
Why? because `bpf_probe_read_user_str` returns positive on success?

Let's fix it in other call sites as well? This may be the source of other errors perhaps?

Reply:
A positive result is ok because we specifically check < 0 not != 0. The point here is not to intermix the string length with an error code that is or'ed later.

Reply:
I see. And this case doesn't repeat with other `read_str` calls?

Reply:
Added checks.